### PR TITLE
Minor layout changes

### PR DESCRIPTION
**Pull request by danilobassi8**
This PR does:
- Add a observable that emits the user in the auth service. This way we can react to user changes and change things in the layout.
- Change the last item in the menu that react to user changes.
image
image
- add a link to the register view
image

- fixes password validation in the signIn
- add a modal that shows how points are assigned to users.
image

### CODE REVIEW

File: src/app/components/pre-mundial-predictions/pre-mundial-predictions.component.html [hunk 1]
```diff
@@ -56,4 +64,54 @@
...
+        <th scope="row" style="width: 15ex;">4 puntos</th>
+      </tr>
+      <tr>
+        <td scope="row">Por cada equipo semi finalista</td>
+        <th scope="row" style="width: 15ex;">8 puntos</th>
+      </tr>
+      <tr>
+        <td scope="row">Por cada equipos compitiendo por el 3er puesto</td>
+        <th scope="row" style="width: 15ex;">12 puntos</th>
+      </tr>
```
Comment (srenzi) [inline src/app/components/pre-mundial-predictions/pre-mundial-predictions.component.html:106]: You're missing the points for _finalist_ which are 20 points.

File: src/app/components/layout/layout.component.html [hunk 2]
```diff
@@ -9,10 +9,37 @@
...
     </ng-template>
+    <div *ngIf="this.auth.currentUser | async as user; else LoginBtn" class="mr-2">
+      <button
+        class="p-button-outlined max-content"
+        type="button"
+        pButton
+        icon="pi pi-user"
+        [label]="user.first_name + ' ' + user.last_name"
+        (click)="menu.toggle($event)"
+      ></button>
```
Comment (srenzi) [inline src/app/components/layout/layout.component.html:20]: Can we make this button the default so that I can hit _Enter_ to login? :pray:

Comment (danilobassi8) [reply src/app/components/layout/layout.component.html:20]: Done!

File: src/app/components/pre-mundial-predictions/pre-mundial-predictions.component.html (hunk 1 above)
Comment (danilobassi8) [reply src/app/components/pre-mundial-predictions/pre-mundial-predictions.component.html:106]: Added!

**Review by srenzi — APPROVED** [review]

**Pull request merged**